data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java

```diff
@@ -0,0 +1,137 @@
+import software.amazon.awssdk.services.s3.model.ListObjectsResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.S3Object;
+@ExtendWith(MockitoExtension.class)
+class S3SinkServiceIT {
+
+    private static final Logger LOG = LoggerFactory.getLogger(S3SinkServiceIT.class);
+    private S3Client s3Client;
+    private String bucketName;
+    private JsonCodec codec;
+    private BufferFactory bufferFactory;
+    @Mock
+    private S3SinkConfig s3SinkConfig;
+    @Mock
+    private ThresholdOptions thresholdOptions;
+    @Mock
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+    @Mock
+    private ObjectKeyOptions objectKeyOptions;
+    @Mock
+    private BucketOptions bucketOptions;
+
+    @BeforeEach
+    public void setUp() {
+        String s3region = System.getProperty("tests.s3ink.region");
+
+        s3Client = S3Client.builder().region(Region.of(s3region)).build();
+        bucketName = System.getProperty("tests.s3sink.bucket");
+
+        codec = new JsonCodec();
+        bufferFactory = new InMemoryBufferFactory();
+
+        when(objectKeyOptions.getNamePattern()).thenReturn("my-elb-%{yyyy-MM-dd'T'hh-mm-ss}");
+        when(bucketOptions.getBucketName()).thenReturn(bucketName);
+        when(bucketOptions.getObjectKeyOptions()).thenReturn(objectKeyOptions);
+        when(thresholdOptions.getEventCount()).thenReturn(10);
+        when(thresholdOptions.getMaximumSize()).thenReturn(ByteCount.parse("2mb"));
+        when(thresholdOptions.getEventCollectTimeOut()).thenReturn(Duration.parse("PT3M"));
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getBucketOptions()).thenReturn(bucketOptions);
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of(s3region));
+
+    }
+
+    @Test
+    void verify_flushed_records_into_s3_bucket() {
+
+        int s3ObjectCountBeforeIngest = gets3ObjectCount();
+        S3SinkService s3SinkService = createObjectUnderTest();
+        s3SinkService.output(setEventQueue());
+        int s3ObjectCountAfterIngest = gets3ObjectCount();
+        assertThat(s3ObjectCountAfterIngest, equalTo(s3ObjectCountBeforeIngest + 1));
+    }
+
+    private S3SinkService createObjectUnderTest() {
+        return new S3SinkService(s3SinkConfig, bufferFactory, codec);
+    }
+
+    private int gets3ObjectCount() {
+        int s3ObjectCount = 0;
+        try {
+            ListObjectsRequest listObjects = ListObjectsRequest.builder().bucket(bucketName).build();
+            ListObjectsResponse res = s3Client.listObjects(listObjects);
+            List<S3Object> objects = res.contents();
+            s3ObjectCount = objects.size();
+        } catch (S3Exception e) {
+            LOG.error(e.awsErrorDetails().errorMessage());
+            System.exit(1);
```

Comment:
This is going to fail the whole test suite and will prevent accurate reporting of failures.

Just let the exception be thrown here. Remove the try-catch block entirely. JUnit will handle it.

Reply:
Removed the try-catch block.